src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/__tests__/TargetingRulesTab.test.tsx

```diff
@@ -125,6 +125,28 @@ describe('TargetingRulesTab', () => {
       expect(onVariationDropdown).toHaveValue('False')
     })
 
+    test('it should show the confirm modal when trying to save an altered default rule, that affects the current flag state', async () => {
+      renderComponent()
+
+      const onVariationDropdown = document.querySelector('input[name="onVariation"]') as HTMLSelectElement
+      expect(onVariationDropdown).toHaveValue('True')
+      await userEvent.click(onVariationDropdown)
+
+      const onVariationDropdownOptions = document.querySelectorAll('li')
+      expect(onVariationDropdownOptions).toHaveLength(2)
```

Comment:
Try to use `screen.getAllByRole` instead
```suggestion
      const onVariationDropdownOptions = screen.getAllByRole('listitem')
      expect(onVariationDropdownOptions).toHaveLength(2)
```

Reply:
Have updated